[Feat] 아이디 찾기 및 아이디 찾기 결과 페이지 디자인 적용

Jone4865 opened the pull request:
## 어떤 이유로 PR을 하셨나요?

- [X] feature 병합
- [ ] 버그 수정
- [X] 코드 개선
- [ ] 기타(아래에 자세한 내용을 기입해주세요)

## 스크린샷 및 세부 내용 - 왜 해당 PR이 필요한지 자세하게 설명해주세요
- [Design] 아이디 찾기 페이지를 공통컴포넌트를 사용하여 재구현 하였습니다.
- [Design] 아이디 찾기 결과 페이지에 반응형 디자인을 적용하였습니다.
- [Feat] 아이디 찾기 api를 연동 하였습니다.


https://github.com/user-attachments/assets/a9d11fd6-0408-4ffb-a62b-d3bbdedbf9ef

Jone4865 committed:
Design
 1. 아이디 찾기 페이지 디자인 재구현
 2. 아이디 찾기 결과페이지 반응형 디자인 적용

Jone4865 committed:
feat : 아이디 찾기 api 연동

Jone4865 committed:
fixed : 컴포넌트 누락 반영

Jone4865 committed:
Refactor
1. FC 컴포넌트 제거
2. padding bottom 16px로 맞춤
3. id 찾기 페이지 button disable 추가
4. id 찾기 result page 디자인 수정 및 버튼 액션 추가

Jone4865 committed:
Refactor
1. id 찾기 요청시 분기점 처리
2. 에러코드 추가

Jone4865 committed:
Fix
1. params 조회 부분 Suspense 안에 적용

Jone4865 committed:
Merge branch 'develop' into design/find-id

heum2 commented on src/app/(auth)/find-id/components/Form/index.tsx:
```diff
@@ -0,0 +1,91 @@
+      try {
+        const response = await postFindId({ email, nickname });
+        router.push(
+          CLIENT_APP_ROUTES.FIND_ID +
+            `/result?nickname=${nickname}&memberId=${response.memberId}`,
```
P1: `FIND_ID_RESULT`를 만드신 것 같은데, 해당 `constants` 이용해주시면 감사드리겠습니다.

heum2 commented on src/app/(auth)/find-id/components/Form/index.tsx:
```diff
@@ -0,0 +1,91 @@
+    >
+      <div className="flex flex-col gap-5">
+        <div className="flex flex-col gap-[10px]">
+          <h2 className="text-[#505050] text-[14px]">닉네임</h2>
```
P3: 해당 줄은 lable로 보이는데, `#505050` 색상을 `tailwind.config.ts` 폴더에서 textColor에 label로 하나 만들면 어떨까요?

Jone4865 committed:
Fix
1. 가로 padding 20 > 16으로 통일

heum2 commented on src/app/(auth)/find-id/result/page.tsx:
```diff
@@ -1,73 +1,58 @@
 
-const ResultPageData: React.FC = () => {
-  const router = useRouter();
+import { Suspense } from 'react';
```
P1: 사용하지 않는 `Suspense` 는 제거해주세요.

heum2 replied:
제거가 안된거 같아요. 확인 부탁드려요~

heum2 replied:
P1: 제거가 안된거 같아요. 확인 부탁드려요~

heum2 commented on src/app/(auth)/find-id/result/page.tsx:
```diff
@@ -1,73 +1,58 @@
-        </g>
-      </svg>
-    </div>
+      <Button onClick={onClcickHandle}>로그인</Button>
```
P1: 로그인 페이지로 이동만 하는 기능이라면, `Link` 를 이용해주세요.
```suggestion
      <Link href={CLIENT_APP_ROUTES.LOGIN}>
        <Button>로그인</Button>
      </Link>
```

heum2 commented on src/constants/routes.ts:
```diff
@@ -5,13 +5,18 @@ export const API_ROUTES = {
     SIGN_UP: `${BASE_API_URL}/api/member/signUp`,
     SIGN_IN: `${BASE_API_URL}/api/member/signIn`,
   },
+  FIND: {
+    FIND_ID: `${BASE_API_URL}/api/member/findMemberId`,
+    FIND_PASSWORD: `${BASE_API_URL}/api/member/findPassword`,
```
P2: 위 방식이면 `API_ROUTES.FIND.FIND_ID` 로 사용이 될텐데, 어떤 API 에서 왔는지 `routes.ts`를 한번 더 들어가야하는 수고스러움이 있을 것 같아요. 
`API_ROUTES`를 보고 바로 스웨거에서 확인할 수 있게 `member` 관련 API 는 `member`로 합치는 것이 어떤가요?

Jone4865 committed:
Refactor
1. 상수 최대한 활용 적용
2. label 부분 #505050 기본 text-icon 적용
3. 로그인 페이지로 이동 버튼 next Link 적용
4. 가독성 위해 FIND_ID, FIND_PASSWORD Member 단에 추가

Jone4865 committed:
Refactor
1. useRoute 미사용으로 제거

Jone4865 committed:
Refactor : 미사용 Suspense 제거

Jone4865 committed:
Fix : Suspense 에러 수정

heum2 reviewed (approved):
수고하셨습니다. 👍
